backends/tfhe-cuda-backend/cuda/src/device.cu

```diff
@@ -52,184 +50,109 @@ void *cuda_malloc_async(uint64_t size, cuda_stream_t *stream) {
+    abort();
   }
 
-  cudaSetDevice(stream->gpu_index);
+  check_cuda_error(cudaSetDevice(stream->gpu_index));
   check_cuda_error(
       cudaMemcpyAsync(dest, src, size, cudaMemcpyHostToDevice, stream->stream));
-  return 0;
 }
 
-/// Tries to copy memory to the GPU synchronously
-/// 0: success
-/// -1: error, invalid device pointer
-/// -2: error, gpu index doesn't exist
-/// -3: error, zero copy size
-int cuda_memcpy_to_gpu(void *dest, void *src, uint64_t size) {
-  if (size == 0) {
-    // error code: zero copy size
-    return -3;
-  }
-
+/// Copy memory to the GPU synchronously
+void cuda_memcpy_to_gpu(void *dest, void *src, uint64_t size) {
   cudaPointerAttributes attr;
-  cudaPointerGetAttributes(&attr, dest);
+  check_cuda_error(cudaPointerGetAttributes(&attr, dest));
   if (attr.type != cudaMemoryTypeDevice) {
-    // error code: invalid device pointer
-    return -1;
+    printf("Cuda error: invalid device pointer in copy to GPU.");
+    abort();
   }
-
   check_cuda_error(cudaMemcpy(dest, src, size, cudaMemcpyHostToDevice));
-  return 0;
 }
 
-/// Tries to copy memory to the CPU synchronously
-/// 0: success
-/// -1: error, invalid device pointer
-/// -2: error, gpu index doesn't exist
-/// -3: error, zero copy size
-int cuda_memcpy_to_cpu(void *dest, void *src, uint64_t size) {
-  if (size == 0) {
-    // error code: zero copy size
-    return -3;
-  }
-
+/// Copy memory to the CPU synchronously
+void cuda_memcpy_to_cpu(void *dest, void *src, uint64_t size) {
   cudaPointerAttributes attr;
-  cudaPointerGetAttributes(&attr, src);
+  check_cuda_error(cudaPointerGetAttributes(&attr, src));
   if (attr.type != cudaMemoryTypeDevice) {
-    // error code: invalid device pointer
-    return -1;
+    printf("Cuda error: invalid device pointer in copy to CPU.");
+    abort();
   }
-
   check_cuda_error(cudaMemcpy(dest, src, size, cudaMemcpyDeviceToHost));
-  return 0;
 }
 
-/// Tries to copy memory within a GPU asynchronously
-/// 0: success
-/// -1: error, invalid device pointer
-/// -2: error, gpu index doesn't exist
-/// -3: error, zero copy size
-int cuda_memcpy_async_gpu_to_gpu(void *dest, void *src, uint64_t size,
+/// Copy memory within a GPU asynchronously
+void cuda_memcpy_async_gpu_to_gpu(void *dest, void *src, uint64_t size,
                                  cuda_stream_t *stream) {
-  if (size == 0) {
-    // error code: zero copy size
-    return -3;
-  }
-
-  if (stream->gpu_index >= cuda_get_number_of_gpus()) {
-    // error code: invalid gpu_index
-    return -2;
-  }
   cudaPointerAttributes attr_dest;
-  cudaPointerGetAttributes(&attr_dest, dest);
+  check_cuda_error(cudaPointerGetAttributes(&attr_dest, dest));
   if (attr_dest.device != stream->gpu_index &&
       attr_dest.type != cudaMemoryTypeDevice) {
-    // error code: invalid device pointer
-    return -1;
+    printf("Cuda error: invalid dest device pointer in copy from GPU to GPU.");
+    abort();
```

Comment:
seeing this get repeated, makes me think we could have panic macro ala Rust
```C++
#include <cstdio>
#include <cstdlib>

#define PANIC(format, ...)                                                     \
  {                                                                            \
    std::fprintf(stderr, "%s::%d::%s: panic.\n" format "\n", __FILE__, __LINE__,    \
            __func__, ##__VA_ARGS__);                                          \
    std::abort();                                                                   \
  }

  int main(int argc, char *argv[])
  {

    PANIC("Hello %s", "world");

    PANIC("oof");
    return 0;
  }
```

Reply:
That's a good idea, I'll do that :pray: